src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -651,6 +660,27 @@ theorem zero_shiftLeft (n : Nat) : 0#w <<< n = 0#w := by
   cases h₁ : decide (i < m) <;> cases h₂ : decide (n ≤ i) <;> cases h₃ : decide (i < n)
   all_goals { simp_all <;> omega }
 
+theorem shiftLeft_xor_distrib (x y : BitVec w) (n : Nat) :
+    (x ^^^ y) <<< n = (x <<< n) ^^^ (y <<< n) := by
+  ext i
+  simp
```

Comment:
It's a bit of an edge-case, but it might be good to edge on the safe side and squeeze this simp (idem ditto in the next two proofs)

Reply:
Done.